network/tests/test_linux.py

```diff
@@ -0,0 +1,338 @@
+        return None
+    elif location == '/sys/class/net/invalid/tx_queue_len':
+        return None
+
+
+@mock.patch('datadog_checks.network.network.Platform.is_linux', return_value=True)
+def test_returns_the_right_instance(is_linux, check):
+    check_instance = check({})
+    assert isinstance(check_instance, LinuxNetwork)
+
+
+@pytest.mark.skipif(not Platform.is_linux(), reason="Only works on Linux systems")
+def test_collect_cx_queues(check, aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    instance['collect_connection_queues'] = True
+    instance['collect_connection_state'] = True
+    check_instance = LinuxNetwork('network', {}, [instance])
+
+    check_instance.check({})
+
+    for metric in CONNECTION_QUEUES_METRICS + common.EXPECTED_METRICS:
+        aggregator.assert_metric(metric)
+
+
+@pytest.mark.skipif(not Platform.is_linux(), reason="Only works on Linux systems")
+def test_collect_cx_queues_when_ss_fails(check, aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    instance['collect_connection_queues'] = True
+    instance['collect_connection_state'] = True
+    check_instance = LinuxNetwork('network', {}, [instance])
+    with mock.patch('datadog_checks.network.check_linux.get_subprocess_output') as out:
+        out.side_effect = ss_subprocess_mock_fails
+        check_instance.check({})
+
+    for metric in CONNECTION_QUEUES_METRICS + common.EXPECTED_METRICS:
+        aggregator.assert_metric(metric)
+
+
+@pytest.mark.skipif(Platform.is_windows(), reason="Only runs on Unix systems")
+def test_cx_state(aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    instance['collect_connection_state'] = True
+    check_instance = LinuxNetwork('network', {}, [instance])
+
+    with mock.patch('datadog_checks.network.check_linux.get_subprocess_output') as out:
+        out.side_effect = ss_subprocess_mock
+        check_instance.check(instance)
+        for metric, value in iteritems(CX_STATE_GAUGES_VALUES):
+            aggregator.assert_metric(metric, value=value)
+        aggregator.reset()
+
+        out.side_effect = netstat_subprocess_mock
+        check_instance.check(instance)
+        for metric, value in iteritems(CX_STATE_GAUGES_VALUES):
+            aggregator.assert_metric(metric, value=value)
+
+
+@pytest.mark.skipif(Platform.is_windows(), reason="Only runs on Unix systems")
+@mock.patch('os.listdir', side_effect=os_list_dir_mock)
+@mock.patch('datadog_checks.network.check_linux.LinuxNetwork._read_int_file', side_effect=read_int_file_mock)
+def test_linux_sys_net(listdir, read_int_file, aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    check_instance = LinuxNetwork('network', {}, [instance])
+
+    check_instance.check({})
+
+    for metric, value in iteritems(LINUX_SYS_NET_STATS):
+        aggregator.assert_metric(metric, value=value[0], tags=['iface:lo'])
+        aggregator.assert_metric(metric, value=value[1], tags=['iface:ens5'])
+    aggregator.reset()
```

Comment:
Is this resetting necessary a the end of a test? (I know, this was there already)

Reply:
No https://github.com/DataDog/integrations-core/blob/cc3f9983e71b4a410b3d896650162c3b2de8d246/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py#L47